lib/iris/fileformats/netcdf.py

```diff
@@ -2244,13 +2284,28 @@ def is_valid_packspec(p):
                 raise ValueError(msg)
         packspecs = packing
 
+    if isinstance(fill_value, six.string_types):
```

Comment:
This whole block is surprising in its data-validation. Is it consistent with other parts of this codebase? If not, what is the motivation for adding such complexity?

Reply:
I need to be handle a single value or iterable of values which may be strings. If it's a single value it needs to be converted to an iterable that can be `zip`ped with `cube` if `cube` is a list. I couldn't find a more succinct way of achieving that, though maybe there is.

Reply:
Can you not use np.array(..., ndmin=1) for this ?
E.G.
```
>>> for thing in ('a', ('a', 'bc'), ['ab', 'c'], 1.0, [1, 2], np.array([1., 2])):
...   print thing, np.array(thing, ndmin=1)
... 
a ['a']
('a', 'bc') ['a' 'bc']
['ab', 'c'] ['ab' 'c']
1.0 [ 1.]
[1, 2] [1 2]
[ 1.  2.] [ 1.  2.]
>>> 
```
It does the special-casing of strings for you, and handles all the iterables uniformly.
I'm not sure about the repeat operation though.

Reply:
Nice suggestion.

Reply:
Yeah I was playing around with it but it doesn't work for generators.

Reply:
I've finally accepted that this *can't* be improved on, though it is painful !
However, could you add a comment to say exactly what all this is about ?
? Maybe something like ...
```
    # Make full-value(s) into an iterable over cubes.
    if isinstance(fill_value, six.string_types):
        # Strings are awkward -- handle separately.
        fill_values = repeat(fill_value)
    else:
        ...
```